specification/http/src/test/java/org/kaazing/specification/http/rfc7231/RequestMethodsIT.java

```diff
@@ -15,39 +15,78 @@
  */
 package org.kaazing.specification.http.rfc7231;
 
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
 import org.junit.Ignore;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
 
 /**
  * Test to validate behavior as specified in <a href="https://tools.ietf.org/html/rfc7231#section-4">RFC 7231 section 4:
  * Request Methods</a>.
  */
 public class RequestMethodsIT {
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/http/rfc7231/method.definitions");
+    private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(timeout);
 
     @Test
-    @Ignore("Not Implemented")
-    public void serverMustRespondToUnknownMethodWith501() {
+    @Specification({
+        "501/request",
+        "501/response" })
+    public void serverMustRespondToUnknownMethodWith501() throws Exception {
         // When a request method is received
         // that is unrecognized or not implemented by an origin server, the
         // origin server SHOULD respond with the 501 (Not Implemented) status
         // code.
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "get/request",
+        "get/response" })
+    public void serverShouldImplementGet() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "head/request",
+        "head/response" })
+    public void serverShouldImplementHead() throws Exception {
+        k3po.finish();
     }
 
     @Test
-    @Ignore("Not Implemented")
-    public void serverShouldImplementGet() {
-        // General purpose servers should implement GET and HEAD
+    @Specification({
+        "post/request",
+        "post/response" })
+    public void serverShouldImplementPost() throws Exception {
+        k3po.finish();
     }
 
     @Test
-    @Ignore("Not Implemented")
-    public void serverShouldImplementHead() {
-        // General purpose servers should implement GET and HEAD
+    @Specification({
+        "put/request",
+        "put/response" })
+    public void serverShouldImplementPut() throws Exception {
+        k3po.finish();
     }
 
     @Test
-    @Ignore("Not Implemented")
-    public void serverShouldRespondWith405ToUnrecognizedMethods() {
-        // 405 (Method Not Allowed)
+    @Specification({
+        "405/request",
+        "405/response" })
```

Comment:
We should rename this test to more properly reflect the RFC, i.e change Unrecognized to "NotAllowed" or something like that.

>When a request method is received that is known by an origin
   server but not allowed for the target resource, the origin server
   SHOULD respond with the 405 (Method Not Allowed) status code.